pieceOfCake5_2.py

```diff
@@ -0,0 +1,10 @@
+def get_recipe_price(prices,optionals=None,**ingredients):
+  pricesCopy=copy.deepcopy(prices)
+  if optionals:
+    for optional in optionals:
+      pricesCopy.pop(optional)
+  sum=0
+  for ingredient, amount in ingredients.items():
+    sum+=prices[ingredient]*amount/100
```

Comment:
After you deep copied prices to ignore the optional ingridients, it appears that you are end using prices and not the pricesCopy :)